### PR TITLE
test: nicer matchers output

### PR DESCRIPTION
**Pull request by aslushnikov**
This patch improves matchers to provide more information once
some of them fails.

### CODE REVIEW

File: utils/testrunner/Matchers.js [hunk 1]
```diff
@@ -43,50 +43,60 @@ class Expect {
...
       const result = matcher.call(null, value, ...args);
-      const message = result.message || `expect.${matcherName} failed`;
+      const message = `expect.${matcherName} failed` + (result.message ? `: ${result.message}` : '');
       console.assert(result.pass !== inverse, message);
     }
   }
 }
 
 const DefaultMatchers = {
   toBe: function(value, other, message) {
```
Comment (JoelEinbinder) [inline utils/testrunner/Matchers.js:53]: If you want to be fancy
function (value, other, message=`${value} == ${other}`)

File: utils/testrunner/Matchers.js [hunk 2]
```diff
@@ -43,50 +43,60 @@ class Expect {
...
     return { pass: value === null, message };
   },
 
   toContain: function(value, other, message) {
+    message = message || `${value} ⊇ ${other}`;
     return { pass: value.includes(other), message };
   },
 
   toEqual: function(value, other, message) {
+    message = message || `Object::${value.constructor.name} ≈ Object::${other.constructor.name}`;
```
Comment (JoelEinbinder) [inline utils/testrunner/Matchers.js]: Jasmine outputs json here. We use toEqual with plane objects, so the constructor name doesn't help much.

File: utils/testrunner/Matchers.js (hunk 1 above)
Comment (aslushnikov) [reply utils/testrunner/Matchers.js:53]: nice! I'll try it out some other time

**Review by JoelEinbinder — APPROVED** [review]

File: utils/testrunner/Matchers.js (hunk 2 above)
Comment (aslushnikov) [reply utils/testrunner/Matchers.js]: I remember huge JSON.stringifed output of `Request` objects in our tests from time to time which I hated, but it seems to be never a case for `toEqual` matcher. Switched to `JSON.stringify` here.

**Pull request merged**